### PR TITLE
Z score library added

### PR DESCRIPTION
**Pull request by prajurock**
**Task Completed By Library**
Z score of weight vs Age
Z score of Height vs Age
Z score of Weight vs length

Malnutrition class of children (Verified by Jana(ACF) and Shivangi)

Z score with library

Growth chart for Nutritionist
a) Growth chart of Weight vs Age
b) Growth chart of Height vs Age
c) Growth chart of Weight vs Height

**_Unit Test Case written_**

TODO:
1.Update Readme in Pypi
2.Code feedback
3.Build Fail if any
4.Licence?

### CODE REVIEW

**Comment by prajurock** [comment]
https://peditools.org/growthwho/
Tool for z score

**Comment by prajurock** [comment]
[Malnutrition Classification.docx](https://github.com/Welthungerhilfe/cgm-ml/files/5327257/Malnutrition.Classification.docx)
Flow chart for Pop up msg - Page 2 (1)
CGM Flow Chart (1)

File: src/common/zscore/LICENSE [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Copyright (c) UNICEF and individual contributors.
```
Comment (markus-hinsche) [inline src/common/zscore/LICENSE:1]: This repository already has a license. Should we remove this? Or how is this license related?

File: src/common/zscore/README [hunk 2]
```diff
@@ -0,0 +1,32 @@
+Z score
+
+z-scores of anthropometric measurements of children below 5 years  based on WHO
```
Comment (markus-hinsche) [inline src/common/zscore/README:3]: Please write a sentence, because this is easier to understand

File: src/common/zscore/README [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+v=Calculator().zScore_wfa(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to calculate z score for weight vs age
+v=Calculator().zScore_wfl(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to calculate z score for weight vs length/height
+v=Calculator().zScore_wfh(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to calculate z score for weight vs length/height
+both wfl and wfh works same
+v=Calculator().zScore_lhfa(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to calculate z score for length vs age
+
+v=Calculator().zScore(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to calculate all three z score
+v=calculator.zScore_withclass(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to calculate all three z score along with class
+v=Calculator().SAM_MAM(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')-->Use to find child is SAM/MAM/Healthy
+
```
Comment (markus-hinsche) [inline src/common/zscore/README:29]: Please use the proper markdown syntax for code: https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/creating-and-highlighting-code-blocks

File: .gitignore [hunk 4]
```diff
@@ -130,3 +130,6 @@ venv.bak/
 
 # logs
 *.log
+
+#os
+.DS_Store
```
Comment (markus-hinsche) [inline .gitignore:135]: Please ensure that the last line of code files is empty

File: src/common/zscore/ZscoreWeb/script.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from flask import * 
+from cgmzscore import Calculator
+app = Flask(__name__)  
+  
+@app.route('/')  
+def customer(): 
+   file_path='/Users/prajwalsingh/Desktop/ZScore/ZscoreWeb/image/roob_rbpp_151005.jpg' 
+   return render_template('first.html',user_image=file_path)
```
Comment (markus-hinsche) [inline src/common/zscore/ZscoreWeb/script.py:8]: Please run flake8 to improve code style, e.g. like this: `flake8 src/common/zscore/`

File: src/common/zscore/cgmzscore/ConvertableData/data_convestion_toolkit.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    value = []
+
+    percentaile_data_path = REPO_DIR+'/ConvertableData/'+percentaile_data_path
+    SD_data_path = REPO_DIR+'/ConvertableData/'+SD_data_path
+    output_file_name = REPO_DIR+'/tables/'+output_file_name
+
+    if os.path.exists(output_file_name):
+        os.remove(output_file_name)
+
+    df = pd.read_csv(percentaile_data_path, sep='\t')
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/ConvertableData/data_convestion_toolkit.py:38]:
```suggestion
    df = pd.read_csv(percentile_data_path, sep='\t')
```

File: src/common/zscore/cgmzscore/ConvertableData/data_convestion_toolkit.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+output_file_name = sys.argv[3]
+
+# Day if table is wfa,lhfa
+# Height if table is wfh
+# Lenght if table is wfl
+zscore_type = sys.argv[4]
+
+
+def convertTxtToJSON(percentaile_data_path, SD_data_path, output_file_name, zscore_type):
+    value = []
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/ConvertableData/data_convestion_toolkit.py:29]: Is there a more specific name instead of `value`?

**Review by markus-hinsche — CHANGES_REQUESTED** [review]
Thanks for the PR. I made some comments, mostly about formatting. Feel free to request a review again once you tackled the comments

File: src/common/zscore/cgmzscore/test_cgmzscore.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        v = Calculator().zScore_wfa(weight=str(g), height=str(
+            df['HEIGHT'][i]), sex=sex, age_in_days=str((df['_agedays'][i])))
+        ans = float("{0:.2f}". format(abs(v - df['_ZWEI'][i])))
+        assert ans <= 0.01
+
+
+def test_zScore_lhfa():
+    for i in range(len(df)):
+        if(pd.isnull(df.loc[i, 'HEIGHT']) or pd.isnull(df.loc[i, '_agedays']) or pd.isnull(df.loc[i, '_ZLEN'] or df.loc[i, '_agedays'] == 0)):
+            assert True
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/test_cgmzscore.py:34]: remove because has no effect

File: src/common/zscore/cgmzscore/test_cgmzscore.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
+import pytest
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/test_cgmzscore.py:1]:
```
"""Found code at http://skdjfksdj.com"""
```

File: src/common/zscore/cgmzscore/test_cgmzscore.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+def test_zScore_lhfa():
+    for i in range(len(df)):
+        if(pd.isnull(df.loc[i, 'HEIGHT']) or pd.isnull(df.loc[i, '_agedays']) or pd.isnull(df.loc[i, '_ZLEN'] or df.loc[i, '_agedays'] == 0)):
+            assert True
+            continue
+
+        if df['GENDER'][i] == 1:
+            sex = 'M'
+        else:
+            sex = 'F'
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/test_cgmzscore.py:40]:
```suggestion
        sex = 'M' if df['GENDER'][i] == 1 else 'F'
```

File: src/common/zscore/cgmzscore/test_cgmzscore.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+        v = Calculator().zScore_lhfa(height=str(g), sex=sex,
+                                     age_in_days=str((df['_agedays'][i])))
+
+        print(g)
+
+        ans = float("{0:.2f}". format(abs(v - df['_ZLEN'][i])))
+        assert ans <= 0.01
+
+
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/test_cgmzscore.py:52]:
```
def check_null():
    return pd.isnull(df.loc[i, 'HEIGHT']) or pd.isnull(df.loc[i, 'WEIGHT']) or pd.isnull(df.loc[i, '_agedays']) or pd.isnull(df.loc[i, '_ZWFL'] or df.loc[i, '_agedays'] == 0))
```

File: src/common/zscore/cgmzscore/test_cgmzscore.py [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+def test_zScore_wfh():
+    for i in range(len(df)):
+        if(pd.isnull(df.loc[i, 'HEIGHT']) or pd.isnull(df.loc[i, 'WEIGHT']) or pd.isnull(df.loc[i, '_agedays']) or pd.isnull(df.loc[i, '_ZWFL'] or df.loc[i, '_agedays'] == 0)):
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/test_cgmzscore.py:55]:
```suggestion
        if check_null(): 
```

File: src/common/zscore/cgmzscore/ConvertableData/data_convestion_toolkit.py [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+# Day if table is wfa,lhfa
+# Height if table is wfh
+# Lenght if table is wfl
+zscore_type = sys.argv[4]
+
+
+def convertTxtToJSON(percentile_data_path, SD_data_path, output_file_name, zscore_type):
+    LMS_SD_data = []
+
+    percentile_data_path = REPO_DIR + '/ConvertableData/' + percentile_data_path
```
Comment (markus-hinsche) [inline src/common/zscore/cgmzscore/ConvertableData/data_convestion_toolkit.py:31]:
```suggestion
    percentile_data_path = REPO_DIR / 'ConvertableData' / percentile_data_path
```

**Review by markus-hinsche — CHANGES_REQUESTED** [review]
I added a few more comments and suggestions

File: src/common/zscore/README.rst [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+EXAMPLE USAGE
+=============
+
+calculate z score for weight vs age
+
+.. code-block:: python
+
+    from cgmzscore import Calculator
+
+    v = Calculator().zScore_wfa(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')
```
Comment (markus-hinsche) [inline src/common/zscore/README.rst:34]:
```suggestion
```python
from cgmzscore import Calculator
v = Calculator().zScore_wfa(weight="7.853",muac="13.5",age_in_days='16',sex='M',height='73')
```
```

**Pull request merged**